src/utils/limit_mem.jl

```diff
@@ -0,0 +1,9 @@
+# try working around the CI issues with nightly + 1.10 (#2441) for now:
+if haskey(ENV, "GITHUB_ACTIONS") && VERSION >= v"1.10.0-DEV"
+  # 3.5GB might work on linux, at least sometimes
+  # 10GB should be fine on macos
+  maxmem = Sys.islinux() ? 3.5*(2^30) : 10*(2^30)
+  println("OscarCI: Limiting memory to ", Base.format_bytes(maxmem));
+  @ccall jl_gc_set_max_memory(maxmem::UInt64)::Cvoid
```

Comment:
Wait, does that mean using `jl_gc_set_max_memory` mitigates some of the GC regressions we are seeing? Perhaps that's a useful clue for debugging, @gbaraldi ?